lib/session.js

```diff
@@ -169,15 +163,28 @@ module.exports = class Session {
   }
 
   isModified () {
-    return this[originalHash] !== this[hash]()
+    return Boolean(
+      this[originalHash] !== this[hash]() ||
+      // If maxAge is set, the session is modified.
+      // This is necessary because we don't read the cookie's expiration from the cookie itself.
+      // session.cookie.expires is initially set from Cookie(cookieOpts), so we cannot detect if it changed from maxAge alone.
+      this[maxAge]
+    )
```

Comment:
```suggestion
    if (this[originalHash] !== this[hash]()) {
      return true
    }

    // If maxAge is set, the session is modified.
    // This is necessary because we don't read the cookie's expiration from the cookie itself.
    // session.cookie.expires is initially set from Cookie(cookieOpts), so we cannot detect if it changed from maxAge alone.
    return this[maxAge] != null
```

However, this seems wrong. Just setting a max age should never make `isModified` return `true` every single time

Reply:
>This is necessary because we don't read the cookie's expiration from the cookie itself.

We should probably do that, and isn't that what happens with your `return sess.cookie.expires?.getTime()` in the hash function?

Reply:
Ah -- I'm mistaken. This code is confusing :)

If `rolling` is false, Session.restore will be called, which addresses this.